### PR TITLE
sm/ts-next-corrections

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
fixes TS/ES-next errors from overriding base sfdxCommand org property

### What issues does this PR fix or reference?
@W-9329741@

### CODE REVIEW

**Review by WillieRuemmele — COMMENTED** [review]
will other plugins need to create their own `ensureOrg`? will the upgrades affect `SfdxCommand` in a similar way?

**Comment by mshanemc** [comment]
> will other plugins need to create their own `ensureOrg`? will the upgrades affect `SfdxCommand` in a similar way?

You can normally just add `as Org` in your command if you know it's going to exist and you're merely extending SfdxCommand.

I think this one was weird because it was putting the "requiresUsername" stuff in a dataCommand class and then extending for the actual commands.  You can't override Org at that level of inheritance.

I could have moved all the requiresUsername stuff to all the commands as another option, but this seemed more in line with what Pete was writing (a fat dataCommand that does all the work).

File: packages/plugin-data/src/dataCommand.ts [hunk 1]
```diff
@@ -96,10 +94,17 @@ export abstract class DataCommand extends SfdxCommand {
     return final;
   }
 
+  public ensureOrg(): Org {
+    if (!this.org) {
+      throw new Error('An org is required for a connection');
```
Comment (peternhale) [inline packages/plugin-data/src/dataCommand.ts:99]: @mshanemc I would prefer the error be the same that is emitted when an org cannot be found.

**Review by peternhale — CHANGES_REQUESTED** [review]
@mshanemc what TS error is this addressing?

File: packages/plugin-data/src/dataCommand.ts (hunk 1 above)
Comment (mshanemc) [reply packages/plugin-data/src/dataCommand.ts]: done

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**